service/controller/machinepool/handler/migration/deleteold.go

```diff
@@ -0,0 +1,54 @@
+package migration
+
+import (
+	"context"
+
+	"github.com/giantswarm/apiextensions/v6/pkg/apis/capiexp/v1alpha3"
+	"github.com/giantswarm/microerror"
+)
+
+const (
+	operatorkitMachinePoolExpFinalizer = "operatorkit.giantswarm.io/azure-operator-machine-pool-exp-controller"
+)
+
+func (r *Resource) deleteOldMachinePool(ctx context.Context, oldMachinePool *v1alpha3.MachinePool) error {
```

Comment:
Called from new MachinePool handler, it will deleted old experimental MachinePool.